relayer/msgs.go

```diff
@@ -363,22 +491,26 @@ func (c *Chain) MsgRelayAcknowledgement(counterparty *Chain, packet *relayMsgPac
 		return
 	}
 
-	version := clienttypes.ParseChainID(counterparty.ChainID)
+	if ackRes == nil {
+		return nil, fmt.Errorf("ack packet [%s]seq{%d} has no associated proofs", counterparty.ChainID, packet.seq)
+	}
+
+	version := clienttypes.ParseChainID(c.ChainID)
 	msg := chantypes.NewMsgAcknowledgement(
 		chantypes.NewPacket(
 			packet.packetData,
 			packet.seq,
-			c.PathEnd.PortID,
-			c.PathEnd.ChannelID,
 			counterparty.PathEnd.PortID,
 			counterparty.PathEnd.ChannelID,
+			c.PathEnd.PortID,
+			c.PathEnd.ChannelID,
```

Comment:
this should be set back to how it was. The ack is on the counterparty chain (the receiver). The proof of the ack is being sent to the source chain (the sender). The signer below should be `c`